src/OrchardCore.Modules/OrchardCore.Media.Azure/Services/ImageSharpImageCacheTenantEvents.cs

```diff
@@ -0,0 +1,87 @@
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Microsoft.Extensions.Localization;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Removing;
+using OrchardCore.Modules;
+
+namespace OrchardCore.Media.Azure.Services;
+
+internal class ImageSharpImageCacheTenantEvents : ModularTenantEvents
+{
+    private readonly ImageSharpImageCacheOptions _options;
+    private readonly ShellSettings _shellSettings;
+    private readonly ILogger _logger;
+
+    protected readonly IStringLocalizer S;
+
+    public ImageSharpImageCacheTenantEvents(
+        IOptions<ImageSharpImageCacheOptions> options,
+        ShellSettings shellSettings,
+        ILogger<ImageSharpImageCacheTenantEvents> logger,
+        IStringLocalizer<ImageSharpImageCacheTenantEvents> localizer)
+    {
+        _options = options.Value;
+        _shellSettings = shellSettings;
+        _logger = logger;
+        S = localizer;
+    }
+
+    public override async Task ActivatingAsync()
+    {
+        // Only create container if options are valid.
+        if (_shellSettings.IsUninitialized() ||
+            string.IsNullOrEmpty(_options.ConnectionString) ||
+            string.IsNullOrEmpty(_options.ContainerName) ||
+            !_options.CreateContainer)
+        {
+            return;
+        }
+
+        _logger.LogDebug("Testing Azure Media ImageSharp Image Cache container {ContainerName} existence", _options.ContainerName);
+
+        try
+        {
+            var blobContainer = new BlobContainerClient(_options.ConnectionString, _options.ContainerName);
+            var response = await blobContainer.CreateIfNotExistsAsync(PublicAccessType.None);
+
+            _logger.LogDebug("Azure Media ImageSharp Image Cache container {ContainerName} created.", _options.ContainerName);
+        }
+        catch (RequestFailedException ex)
+        {
+            _logger.LogError(ex, "Unable to create Azure Media ImageSharp Image Cache Container.");
+        }
+    }
+
+    public override async Task RemovingAsync(ShellRemovingContext context)
+    {
+        // Only remove container if options are valid.
+        if (!_options.RemoveContainer ||
+            string.IsNullOrEmpty(_options.ConnectionString) ||
+            string.IsNullOrEmpty(_options.ContainerName))
```

Comment:
should we add this check as a function inside the options class so it is evaluated once and the logic is defined into once place? like this or similar? 
https://github.com/OrchardCMS/OrchardCore/blob/21b746b8f0416985ebc982263d30e830c0c91c2b/src/OrchardCore/OrchardCore.ReCaptcha.Core/Configuration/ReCaptchaSettings.cs#L20